reconcile/utils/pagerduty_api.py

```diff
@@ -1,40 +1,78 @@
-import datetime
+from datetime import datetime as dt
+import logging
+from typing import Callable
+from typing import Iterable
+from typing import Optional
+from typing import Protocol
 
-import requests
 import pypd
+import requests
+from pydantic import BaseModel
+from sretoolbox.utils import retry
 
 from reconcile.utils.secret_reader import SecretReader
+from reconcile.utils.secret_reader import SupportsSecret
+
+
+class PagerDutyTargetException(Exception):
+    ...
+
+
+class PagerDutyInstance(Protocol):
+    name: str
+
+    @property
+    def token(self) -> SupportsSecret:
+        ...
+
+
+class PagerDutyUser(Protocol):
+    org_username: str
+    pagerduty_username: Optional[str]
+
+
+class PagerDutyTarget(Protocol):
+    name: str
+    instance: PagerDutyInstance
+    escalation_policy_id: Optional[str]
+    schedule_id: Optional[str]
+
```

Comment:
What's the reasoning behind this classes being protocols? That we can pass directly objects from `qenerate`? Should that deserve comments/documentation somewhere? 🙏